util/reggen/gen_html.py

```diff
@@ -192,24 +192,31 @@ def gen_html_register(outfile, reg, comp, width, rnames, toc, toclvl):
             ('x' if field.resval is None else hex(field.resval)) +
             "</td>")
         genout(outfile, "<td class=\"regfn\">" + fname + "</td>")
+
+        # Collect up any description and enum table
+        desc_parts = []
+
         if field.desc is not None:
-            genout(outfile, render_td(field.desc, rnames, 'regde'))
-        else:
-            genout(outfile, "<td>\n")
+            desc_parts += expand_paras(field.desc, rnames)
 
         if field.enum is not None:
-            genout(outfile, "    <table>")
+            desc_parts.append('<table>')
             for enum in field.enum:
-                ename = enum.name
-                genout(outfile, "    <tr><td>" + str(enum.value) + "</td>")
-                genout(outfile, "<td>" + ename + "</td>")
-                genout(outfile, render_td(enum.desc, rnames, None))
-                genout(outfile, "</tr>\n")
-
-            genout(outfile, "    </table>")
+                enum_desc_paras = expand_paras(enum.desc, rnames)
+                desc_parts.append('<tr>'
+                                  '<td>{val}</td>'
+                                  '<td>{name}</td>'
+                                  '<td>{desc}</td>'
+                                  '</tr>'
```

Comment:
What's your opinion on inserting a newline for the generated output here? `'</tr>\n'`
For the original problem I looked at the generated output, not having extra long lines and some alignment helps to understand what is happening.

Reply:
Oh, yep, can do.

Reply:
Thanks :)